packages/@yodaos/application/application.js

```diff
@@ -0,0 +1,193 @@
+var path = require('path')
+var _ = require('@yoda/util')._
+var symbol = require('./symbol')
+
+function invoke (target, optionProp, args) {
+  var fn = target[symbol.options][optionProp]
+  if (typeof fn === 'function') {
+    return fn.apply(target, args)
+  }
+}
+
+function fatalError (message) {
+  console.error(message)
+  process.exit(233)
+}
+
+function loadPackage (api, registry) {
+  var appHome = api.appHome
+  var packageJson = require(path.join(appHome, 'package.json'))
+  var services = _.get(packageJson, 'manifest.services', [])
+  var activities = _.get(packageJson, 'manifest.activities', [])
+
+  services.forEach(it => {
+    if (!Array.isArray(it)) {
+      it = [ it ]
+    }
+    var name = it[0]
+    if (typeof name !== 'string') {
+      return
+    }
+    registry.service[name] = {
+      path: path.join(appHome, _.get(it, '1.main', name))
+    }
+  })
+
+  activities.forEach(it => {
+    if (!Array.isArray(it)) {
+      it = [ it ]
+    }
+    var name = it[0]
+    if (typeof name !== 'string') {
+      return
+    }
+    registry.activity[name] = {
+      path: path.join(appHome, _.get(it, '1.main', name))
+    }
+  })
+}
+
+function getComponent (application, name, type) {
+  var registry = application[symbol.registry]
+  var entity = registry[type][name]
+  if (entity == null) {
+    fatalError(`Unknown app component '${type}:${name}'.`)
+    return
+  }
+  var component = entity.mod
+  if (component == null) {
+    component = entity.mod = loadComponent(registry, name, type)
+    component[symbol.componentName] = name
+  }
+  return component
+}
+
+function loadComponent (registry, name, type) {
+  var entity = registry[type][name]
+  var mod = require(entity.path)
+  return mod
+}
+
+function Application (options, api) {
+  if (api == null) {
+    api = global[symbol.api]
+  }
+  var application = api[symbol.application]
+  if (application != null) {
+    throw Error('Multiple calls on Application.')
```

Comment:
throw a function return value of an error?

Reply:
This seems to be an fatal error, so I'm +1 on throwing directly.